From 050d69b99bc1e77537c1de34e916924bd687ef0b Mon Sep 17 00:00:00 2001
From: Limeng <Meng.Li@windriver.com>
Date: Fri, 6 Mar 2020 16:10:22 +0800
Subject: [PATCH] driver: gpu: vc4: pass the correct parameter to function
 drm_hdmi_avi_infoframe_from_display_mode

According to upstream commit 13d0add333af("drm/edid:
Pass connector to AVI infoframe functions"), it is need to pass
struct drm_display_mode as the third argument.

Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/vc4/vc4_firmware_kms.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/vc4/vc4_firmware_kms.c b/drivers/gpu/drm/vc4/vc4_firmware_kms.c
index b9176f073dfd..91dd8956f812 100644
--- a/drivers/gpu/drm/vc4/vc4_firmware_kms.c
+++ b/drivers/gpu/drm/vc4/vc4_firmware_kms.c
@@ -863,7 +863,7 @@ static void vc4_crtc_mode_set_nofb(struct drm_crtc *crtc)
 	union hdmi_infoframe frame;
 	int ret;
 
-	ret = drm_hdmi_avi_infoframe_from_display_mode(&frame.avi, vc4_crtc->connector, false);
+	ret = drm_hdmi_avi_infoframe_from_display_mode(&frame.avi, vc4_crtc->connector, mode);
 	if (ret < 0) {
 		DRM_ERROR("couldn't fill AVI infoframe\n");
 		return;
-- 
2.17.1

